chainstate/src/detail/chainstateref.rs

```diff
@@ -307,8 +304,25 @@ impl<'a, S: BlockchainStorageRead> ChainstateRef<'a, S> {
         self.check_transactions(block)
             .map_err(CheckBlockError::CheckTransactionFailed)?;
 
-        // TODO: Size limits
-        if block.encoded_size() > MAX_BLOCK_WEIGHT {
+        self.check_block_size(block)?;
+
+        Ok(())
+    }
+
+    fn check_block_size(&self, block: &Block) -> Result<(), CheckBlockError> {
+        let block_size = block.block_size();
+
+        if block_size.size_from_header() > self.chain_config.max_block_header_size() {
```

Comment:
I think using `ensure!()` here would be slightly nicer

Reply:
Done